File: CPP/Program/Komponen.cpp
#pragma once
#include <iostream>
#include <string>

using namespace std;

class Komponen {
    private:
        string merk;
        string nama;
    public:
        Komponen(){}

        Komponen(string merk, string nama){
            this->merk = merk;
            this->nama = nama;
        }

        // setter
        void setMerk(string merk){this->merk = merk;}
        void setNama(string nama){this->nama = nama;}

        // getter
        string getMerk(){return merk;}
        string getNama(){return nama;}

        virtual void displayInfo() { // Virtual method
            cout << "Komponen: " << nama << " (" << merk << ")" << endl;
        }

        virtual ~Komponen() {}

};

File: Java/Program/Komponen.java

public class Komponen {
    private String merk;
    private String nama;
    
    public Komponen() {}
    
    public Komponen(String merk, String nama) {
        this.merk = merk;
        this.nama = nama;
    }
    
    // Setter methods
    public void setMerk(String merk) {this.merk = merk;}
    public void setNama(String nama) {this.nama = nama;}
    
    // Getter methods
    public String getMerk() {return merk;}
    public String getNama() {return nama;}
    
    // Virtual method 
    public void displayInfo() {
        System.out.println("Komponen: " + nama + " (" + merk + ")");
    }
    
    
}